lib/diaspora/exporter/posts_with_activity.rb

```diff
@@ -0,0 +1,51 @@
+module Diaspora
+  class Exporter
+    # This class allows to query posts where a person made any activity (submitted comments,
+    # likes, participations or poll participations).
+    class PostsWithActivity
+      # TODO: docs
+      def initialize(person)
+        @person = person
+      end
+
+      # TODO: docs
+      def query
+        Post.from("(#{sql_union_all_activities}) AS posts")
+      end
+
+      private
+
+      attr_reader :person
+
+      def sql_union_all_activities
+        all_activities.map(&:to_sql).join(" UNION ")
+      end
+
+      def all_activities
+        [comments_activity, likes_activity, subscriptions, polls_activity].compact
+      end
+
+      def likes_activity
+        other_people_posts.liked_by(person)
+      end
+
+      def comments_activity
+        other_people_posts.commented_by(person)
+      end
+
+      def subscriptions
+        return nil if person.owner.nil?
```

Comment:
This should never be nil, because we can only export a users data. And I think I would just change `person` to `user` in the constructor to make that clear.